scripts/depsvalidator/README.md

```diff
@@ -0,0 +1,100 @@
+# DepsValidator
+
+DepsValidator is a command line utility that can ensure that a library is using
+a consistent set of dependency versions across multiple Apple-ecosystem package
+managers.
+
+It can validate dependencies specified according to Semantic Versioning with:
+
+- Swift Package Manager
+- CocoaPods
+- Carthage
+- JSON files containing an object with dependency names as keys and the
+  corresponding pinned versions as values.
+
+## Usage
+
+To use DepsValidator, you need to write a config file that specifies the paths
+to your dependency managers' manifests and lists the dependencencies. By
+default, the `depsvalidator` command will look for the config file at
+`.depsvalidator.yml` in the current directory. You may specify a different file
+using the `--config-file` option.
+
+### Config File Structure
+
+The config file has the following structure:
+
+```yaml
+---
+manifests:
+  - type: [Package.swift, Package.resolved, Podspec, Cartfile, Cartfile.resolved, JSON]
+    path: {path to manifest}
+    omit_for:
+      - {dependency name}
+dependencies:
+  - name: {dependency name}
+    variations:
+      [SPM, CocoaPods, Carthage, JSON]: {name variation}
+```
+
+- `manifests`: An array of manifest declarations.
+  - `type`: The type of manifest. Must be one of the values listed above.
+  - `path`: [optional] The path to the manifest, relative to the path of the
+    config file. By default, depsvalidator will look in the same directory as
+    the config file. The default path for JSON manifests is `versions.json`.
+  - `omit_for`: [optional] An array of dependency names which should not be
+    expected in the manifest. The name must match the name of one of the
+    dependencies listed in this config file.
+- `dependencies`: An array of dependency declarations.
+  - `name`: The name of the dependency.
+  - `variations`: a dictionary that can be used to override the name of the
+    dependency for specific package managers. Keys are one of the values shown
+    above. Values are the correct name for that package manager.
+
+It is permissible for there to be more than one manifest of a given type. For
+example, if your repository hosts a Swift package and an Xcode workspace that
+uses Swift packages, you'll likely have two Package.resolved files (one in the
+repo root, and one at
+`YourWorkspace.xcworkspace/xcshareddata/swiftpm/Package.resolved`).
+
+### Command Line Invocation
+
+You can build and test DepsValidator using the standard Swift Package Manager
+command line techniques, including `swift build` and `swift test`. You can also
+run it directly using `swift run depsvalidator {subcommand} {arguments}`.
```

Comment:
I think this would be a good place to add a note/advise about the best way to use this script in CI